src/Symfony/Component/Notifier/Bridge/KazInfoTeh/KazInfoTehTransport.php

```diff
@@ -0,0 +1,115 @@
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Egor Taranov <dev@taranovegor.com>
+ */
+class KazInfoTehTransport extends AbstractTransport
+{
+    protected const HOST = 'kazinfoteh.org';
+
+    private string $username;
+
+    private string $password;
+
+    private string $sender;
+
+    public function __construct(string $username, string $password, string $sender, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->sender = $sender;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf(
+            'kazinfoteh://%s?sender=%s',
+            $this->getEndpoint(),
+            $this->sender
+        );
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage
+            && 0 === strpos($message->getPhone(), '77')
+            && 11 === \strlen($message->getPhone())
+        ;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage || !$this->supports($message)) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf('http://%s/api', $this->getEndpoint());
+        $response = $this->client->request('POST', $endpoint, [
+            'query' => [
+                'action' => 'sendmessage',
+                'username' => $this->username,
+                'password' => $this->password,
+                'recipient' => $message->getPhone(),
+                'messagetype' => 'SMS:TEXT',
+                'originator' => $this->sender,
+                'messagedata' => $message->getSubject(),
+            ],
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote KazInfoTeh server.', $response, 0, $e);
+        }
+
+        $content = $response->getContent(false);
+        if (200 !== $statusCode || false === strpos($content, '<statuscode>0</statuscode>')) {
+            // <statusmessage>There's any text</statusmessage> => [0 => There's any text]
+            $pattern = 0 !== preg_match('#<statuscode>\d+<\/statuscode>#m', $content)
```

Comment:
please use the proper XML parsing available in PHP instead of using a regex-based logic (which is not handling escaping properly for instance)

Reply:
Can we use `ext-simplexml`?

Reply:
yes